FluidNC/src/Kinematics/GenericCartesian.h

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020 -	Bart Dring
+// Copyright (c) 2023 -	Vlad A.
+// Use of this source code is governed by a GPLv3 license that can be found in the LICENSE file.
+
+#pragma once
+
+/*
+	GenericCartesian.h
+
+	This is a kinematic system for where the motors operate in the cartesian space.
+
+    An abstract class which should be a base for diferent transformation cases.
+
+    Unlike Cartisian kinamatic system, this system allow motor space to vary from machine space. 
+	Transformation from system space to motor space is defined as matrix.
+    All axis must by linearly independent. Matrix has to be inversable.
+
+*/
+
+#include "Kinematics.h"
+
+#define LOG_MATRIX_CONTENT
+
+namespace Kinematics {
+
+    class GenericCartesian : public KinematicSystem {
+    public:
+        GenericCartesian() = default;
+
+        GenericCartesian(const GenericCartesian&)            = delete;
+        GenericCartesian(GenericCartesian&&)                 = delete;
+        GenericCartesian& operator=(const GenericCartesian&) = delete;
+        GenericCartesian& operator=(GenericCartesian&&)      = delete;
+
+        // Kinematic Interface
+        virtual void init() override;
+        virtual void init_position() override;
+
+        virtual bool cartesian_to_motors(float* target, plan_line_data_t* pl_data, float* position) override;
+        void         motors_to_cartesian(float* cartesian, float* motors, int n_axis) override;
+        void         transform_cartesian_to_motors(float* cartesian, float* motors) override;
+
+        bool canHome(AxisMask axisMask) override;
+        void releaseMotors(AxisMask axisMask, MotorMask motors) override;
+        bool limitReached(AxisMask& axisMask, MotorMask& motors, MotorMask limited) override;
+
+        // Configuration handlers:
+//        void afterParse() override {}
+//        void group(Configuration::HandlerBase& handler) override;
+//        void validate() override {}
+
+        // Name of the configurable. Must match the name registered in the cpp file.
+//        const char* name() const override { return "Generic Cartesian"; }
+
+    protected:
+        template <typename number>
+        class Mtx {
+            uint    _pitch;
```

Comment:
Well, this is the whole point, to decouple matrix geometry, from memory layout, and make sure no one confuse one term and another. Pitch is equal to column number in this particular case, but this may be not always true. 
This is very common source of mistakes, when people use the same naming which starts to imply the way data stored in memory. Pitch always define an offset between lines and line can store whatever. In general matrices can be stored in column-major or row-major order. And it's really hard to predict which one will be more optimal and then it's hard to change it when all variables related to storage has naming related to matrix geometry. 

And yes, I know what you'll say, that this is not relevant to ESP32, where memory layout is almost irrelevant. And I'll answer, that, yes you're right. I just cannot get rid of habit to name things the way it should be. 

Sure, I can rename it, if you insist.